server/core/src/main/java/io/whitefox/core/services/IcebergCatalogHandler.java

```diff
@@ -0,0 +1,70 @@
+package io.whitefox.core.services;
+
+import io.whitefox.core.Metastore;
+import io.whitefox.core.MetastoreProperties;
+import io.whitefox.core.Storage;
+import java.io.IOException;
+import java.util.Map;
+import org.apache.iceberg.BaseMetastoreCatalog;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.aws.glue.GlueCatalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.hadoop.HadoopCatalog;
+
+public class IcebergCatalogHandler {
+
+  private final AwsGlueConfigBuilder awsGlueConfigBuilder;
+
+  private final HadoopConfigBuilder hadoopConfigBuilder;
+
+  public IcebergCatalogHandler(
+      AwsGlueConfigBuilder awsGlueConfigBuilder, HadoopConfigBuilder hadoopConfigBuilder) {
+    this.awsGlueConfigBuilder = awsGlueConfigBuilder;
+    this.hadoopConfigBuilder = hadoopConfigBuilder;
+  }
+
+  public Table loadTableWithGlueCatalog(
+      Metastore metastore, Storage storage, TableIdentifier tableIdentifier) {
+    try (var catalog = new GlueCatalog()) {
+      catalog.setConf(hadoopConfigBuilder.buildConfig(storage));
+      catalog.initialize(
+          metastore.name(),
+          awsGlueConfigBuilder.buildConfig(
+              (MetastoreProperties.GlueMetastoreProperties) metastore.properties()));
```

Comment:
is this cast safe?

Reply:
I think yes, because we already checked that the metastore type is "Glue", so we must have the GlueMetastoreProperties in our SharedTable object